addons/web/static/src/core/overlay/overlay_container.js

```diff
@@ -11,6 +11,12 @@ export class OverlayContainer extends Component {
         return sortBy(Object.values(this.props.overlays), (overlay) => overlay.sequence);
     }
 
+    isVisible(overlay) {
+        const rootId =
+            overlay.props.target?.getRootNode()?.host?.id || overlay.props.subProps?.rootId;
+        return rootId === this.env.rootId;
```

Comment:
What about improving all overlays so that they propagate `env`?
I see that `OverlayService.add(component, props, options)` can set proper `env` with providing `options.env`.

If we do that, then this LOC could simply become `overlay.env.rootId`?